src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -12914,14 +13757,48 @@ int Assembler::prefixq_and_encode(int dst_enc, int src_enc) {
   return dst_enc << 3 | src_enc;
 }
 
+int Assembler::prefixq_and_encode_rex2(int dst_enc, int src_enc) {
+  if (dst_enc >= 24) {
```

Comment:
The prefixq_and_encode_rex2 can also be simplified using the above get_base_or_dst_prefixq_bits,  and get_src_prefixq_bits.

Reply:
Done. Thanks.